napari/layers/vectors/vectors.py

```diff
@@ -683,13 +671,7 @@ def _set_view_slice(self):
 
         indices, alphas = self._slice_data(self._slice_indices)
         if not self._dims_displayed == self._displayed_stored:
-            vertices, triangles = generate_vector_meshes(
-                self.data[:, :, list(self._dims_displayed)],
-                self.edge_width,
-                self.length,
-            )
-            self._mesh_vertices = vertices
-            self._mesh_triangles = triangles
+            self._update_mesh()
             self._displayed_stored = copy(self._dims_displayed)
```

Comment:
Should this also be added to `_update_mesh`? It seems like it's mutated every time to keep the `_dims_displayed` last used to index `data`.

Reply:
yes, that's true. done.